### PR TITLE
ENH Show a cannot create message when types is empty

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-linkfield/issues/151

Test with
```php
    public function getCMSFields()
    {
        // ...
        LinkField('MyLink')::create()->setAllowedTypes([MyLink::class]));
        // ...
```

Where MyLink has

```php
    public function canCreate($member = null, $context = [])
    {
        return false;
    }
```

### CODE REVIEW

File: src/Form/Traits/AllowedLinkClassesTrait.php [hunk 1]
```diff
@@ -13,7 +13,9 @@
  */
 trait AllowedLinkClassesTrait
 {
-    private array $allowed_types = [];
+    private array $allowedTypes = [];
```
Comment (emteknetnz) [inline src/Form/Traits/AllowedLinkClassesTrait.php]: Changing the casing in unrelated refactoring

File: client/src/components/LinkPicker/LinkPicker.js [hunk 2]
```diff
@@ -40,12 +40,13 @@ const LinkPicker = ({ types, onModalSuccess, onModalClosed, canCreate }) => {
   const shouldOpenModal = typeKey !== '';
   const className = classnames('link-picker', 'form-control');
   const typeArray = Object.values(types);
+  const message = i18n._t('LinkField.CANNOT_CREATE_LINK', 'Cannot create link');
 
-  if (!canCreate) {
+  if (!canCreate || typeArray.length === 0) {
```
Comment (sabina-talipova) [inline client/src/components/LinkPicker/LinkPicker.js:45]: Probably, we should divide cases when user cannot create link and when there is not available link types.

**Review by sabina-talipova — CHANGES_REQUESTED** [review]
I've tested locally, works perfect. Looks good, just small question.

File: client/src/components/LinkPicker/LinkPicker.js (hunk 2 above)
Comment (emteknetnz) [reply client/src/components/LinkPicker/LinkPicker.js:45]: I'm not sure why we want to have separate logic here? The end result of both of seems like it should be the same, a message saying that you cannot create a link

What else would we do here if we split the logic?

File: src/Form/Traits/AllowedLinkClassesTrait.php [hunk 3]
```diff
@@ -43,16 +46,6 @@ public function getAllowedTypes(): array
      */
     private function validateTypes(array $types): bool
     {
-        if (empty($types)) {
-            throw new InvalidArgumentException(
```
Comment (ChloeSartorelli) [inline src/Form/Traits/AllowedLinkClassesTrait.php]: Still need to throw this exception if developers manually pass in an empty array - see acceptance criteria

Comment (emteknetnz) [reply src/Form/Traits/AllowedLinkClassesTrait.php]: Ah yup my bad, didn't need to change any PHP in the end, have reverted

File: client/src/components/LinkPicker/LinkPicker.js (hunk 2 above)
Comment (sabina-talipova) [reply client/src/components/LinkPicker/LinkPicker.js:45]: No, everything looks good to me.

**Review by sabina-talipova — APPROVED** [review]
LGTM.

**Pull request merged**